### DISCUSSION TITLE
Who's using LDK?

### DISCUSSION THREAD

**Discussion by ConorOkus** [lightningdevkit/rust-lightning#1554] [Show and tell]
If you are using LDK in your project/library/app you can share the URL here.

Some examples:

- Bitkit - https://bitkit.to/
- Mutiny - https://www.mutinywallet.com/
- CashApp - https://cash.app/

**Comment by devrandom** [comment]
Not sure in this context if it make sense in this context to include libraries, like [VLS](https://gitlab.com/lightning-signer/docs/-/blob/master/README.md).

**Reply by ConorOkus** [reply]
Yes I think it should, I reworded my original comment to include project/library

**Reply by moneyball** [reply]
@devrandom which functionality from LDK does VLS use?

**Reply by devrandom** [reply]
Mostly `KeysInterface` (signer interface and `InMemorySigner` implementation) and `chan_utils` (transaction building utilities).

**Comment by dunxen** [comment]
Mobile app still in progress and source code not published yet, but sharing the URL so long so I don't forget: https://umbani.app.

**Reply by moneyball** [reply]
That URL doesn't work for me

**Reply by moneyball** [reply]
URL still doesn't work. @dunxen what's the correct URL?

**Comment by sr-gi** [comment]
The Eye of Satoshi is!  

https://github.com/talaia-labs/rust-teos (I may need to build a fancy website now 😄)

**Comment by shamardy** [comment]
AtomicDEX! https://atomicdex.io/
Lightning integration using LDK is currently in progress.

**Reply by moneyball** [reply]
@shamardy What's the best way to reach you to discuss how your LDK integration is going?

**Reply by shamardy** [reply]
You can reach me on Slack or Discord. Just replied to your message on Slack 🙂

**Comment by AnthonyRonning** [comment]
Working on a few projects with LDK: 

Hidden Lightning Network - https://github.com/BitcoinDevShop/hidden-lightning-network
A privacy focused lightning wallet (based on Sensei/LDK) - https://github.com/BitcoinDevShop/pln
Experimenting with tor by adapting the ldk-sample node - https://github.com/TonyGiorgio/ldk-sample-tor

**Reply by 84adam** [reply]
Hey Tony: Would ldk-sample-tor be useful for standing up a tor-only lightning node based on LDK? Is that the purpose of this project?

**Reply by AnthonyRonning** [reply]
oh hey @84adam , sorry just saw this. Yeah that's the purpose of that one. I'm hoping to learn about the right ways to go about tor in an LDK based project so that I can add it to pln and/or sensei eventually.

**Comment by Bosch-0** [comment]
Would be handy to highlight these projects on the LDK website

**Reply by ConorOkus** [reply]
Agree, a showcase page + case studies have been pending for a while. Let me know if that's something you would be interested in helping with.

**Reply by Bosch-0** [reply]
Yeah sure I could design something

**Comment by MaxFangX** [comment]
Lexe is building a managed, non-custodial wallet using LDK + SGX. We aren't open-source yet, but our GitHub org is here:

https://github.com/lexe-tech

Our landing page will eventually be at either lexe.app or lexe.tech :)

**Comment by Tibo-lg** [comment]
I'm using LDK for rust-dlc, for communication between nodes mainly but also for a PoC of DLC over lightning
https://github.com/p2pderivatives/rust-dlc

**Comment by ConorOkus** [comment]
https://cyberhornet.network/

**Comment by ConorOkus** [comment]
Valera Labs - https://valera.co/

**Comment by Kukks** [comment]
Still hoping for dotnet bindings to be able to use ldk :(

**Reply by tnull** [reply]
Maybe it would be of interest to you that the [LDK Node project](https://github.com/lightningdevkit/ldk-node) (soon to be released) will provide [UniFFI bindings](https://github.com/mozilla/uniffi-rs) and recently a third-party C# bindings generator extension for UniFFI has been announced: https://github.com/NordSecurity/uniffi-bindgen-cs.

I doubt that we'd want to pull in this dependency into LDK Node itself anytime soon, but it should be straight forward to apply this generator yourself once UniFFI bindings support has been merged. Happy to assist with any questions to that end.

**Reply by tnull** [reply]
Also note that there are first experiments with C# bindings for LDK (https://github.com/lightningdevkit/ldk-garbagecollected/tree/main/c_sharp), not sure on any timeline though (cc @TheBlueMatt).

**Reply by Kukks** [reply]
> Maybe it would be of interest to you that the [LDK Node project](https://github.com/lightningdevkit/ldk-node) (soon to be released) will provide [UniFFI bindings](https://github.com/mozilla/uniffi-rs) and recently a third-party C# bindings generator extension for UniFFI has been announced: [NordSecurity/uniffi-bindgen-cs](https://github.com/NordSecurity/uniffi-bindgen-cs).
> 
> I doubt that we'd want to pull in this dependency into LDK Node itself anytime soon, but it should be straight forward to apply this generator yourself once UniFFI bindings support has been merged. Happy to assist with any questions to that end.

Unfortunately, that would be too high-level for us; we would like to implement a node that fits exactly to our needs (our own onchain system, remote signing, magic, etc.) :( 



> Also note that there are first experiments with C# bindings for LDK ([lightningdevkit/ldk-garbagecollected@`main`/c_sharp](https://github.com/lightningdevkit/ldk-garbagecollected/tree/main/c_sharp)), not sure on any timeline though (cc @TheBlueMatt).

Glad to see a recent effort! I recall @joemphilips had done some super early efforts in this regard a while back https://github.com/joemphilips/NRustLightning

**Reply by TheBlueMatt** [reply]
Sadly I think we're a ways away from dotnet bindings. The initial experimentation seemed to show it would be quite doable with our existing infrastructure, but it's a good bit of investment which we don't currently have resources for. Its great to know someone wants it, though, we definitely prioritize based on user needs.

**Comment by benthecarman** [comment]
https://github.com/MutinyWallet/mutiny-node

https://github.com/kuutamolabs/lightning-knd

https://github.com/carlaKC/lndk

**Comment by moneyball** [comment]
Synonym is using LDK in Bitkit https://github.com/synonymdev/react-native-ldk

**Comment by moneyball** [comment]
https://github.com/ConorOkus/uMlando-wallet

**Comment by moneyball** [comment]
C= https://cequals.xyz/

**Comment by moneyball** [comment]
Lipa https://github.com/getlipa

**Comment by moneyball** [comment]
Velas Commerce https://twitter.com/erikmichaelalv1/status/1583482565918281728

**Comment by moneyball** [comment]
Mercury Wallet https://github.com/layer2tech/mercury-wallet/branches

**Comment by moneyball** [comment]
Kuutamolabs https://github.com/kuutamolabs/lightning-knd

**Comment by moneyball** [comment]
CyberHornet Swarm https://cyberhornet.network/

**Comment by moneyball** [comment]
Here's an attempt at categorizing applications building with LDK.

Mobile non-custodial apps:
- Blue Wallet
- Synonym Bitkit
- Lipa
- Velas Commerce
- Portico

Browser non-custodial apps:
- Mutiny

Desktop non-custodial apps:
- CyberHornet Swarm
- Mercury Wallet
- Komodo AtomicDEX
- Hydranet

LSPs:
- c=

Managed non-custodial services:
- Lexe
- Valera
- Voltage

LN routers:
- Kuutamo
- Sensei

Server custodial services:
- Cash App

Services using portions of LDK:
- VLS
- TEOS
- LNDK

DLCs:
- rust-dlc
- 10101 - https://10101.finance/

Smart contract platforms:
- RGB Tools - https://github.com/RGB-Tools

Experimentation:
- Hidden Lightning Network - https://github.com/BitcoinDevShop/hidden-lightning-network
- Experimenting with tor by adapting the ldk-sample node - https://github.com/TonyGiorgio/ldk-sample-tor

Teaching/educational apps:
- uMlando

**Comment by moneyball** [comment]
Portico
https://twitter.com/PorticoExchange
https://porticoexchange.github.io/porticoexchangev2.github.io/

**Comment by moneyball** [comment]
Voltage

**Comment by moneyball** [comment]
Hydranet https://hydranet.ai/

**Comment by johnsBeharry** [comment]
Started a "awesome" directory from this discussion - few projects I wasn't sure how they use LDK though, not personal just didn't know how to categorise them. PRs open though

https://github.com/peakshift/awesome-ldk/

**Comment by crukundo** [comment]
**EttaWallet**: A simple open-source non-custodial bitcoin/lightning wallet with a strong bias towards usability, accessibility and good UX

**Logo**: https://avatars.githubusercontent.com/u/124452068

**Repo**: https://github.com/EttaWallet/EttaWallet

**Comment by EthnTuttle** [comment]
Fedimint is a federated [Chaumian e-cash](https://en.wikipedia.org/wiki/Ecash) mint backed by bitcoin with deposits and withdrawals that can occur on-chain or via Lightning.

https://fedimint.org/

Repo: https://github.com/fedimint/fedimint

Proof: https://github.com/search?q=repo%3Afedimint%2Ffedimint+lightning+lang%3Atoml&type=code

**Reply by moneyball** [reply]
My understanding is that Fedimint has not begun building with LDK yet but they'd like to in the future.

**Reply by EthnTuttle** [reply]
https://github.com/search?q=repo%3Afedimint%2Ffedimint+lightning+language%3ATOML&type=code&l=TOML

<img width="555" alt="Screenshot 2023-05-22 at 6 59 51 PM" src="https://github.com/lightningdevkit/rust-lightning/assets/133234413/22c30d72-e63f-4059-a64f-abd5f4713d94">

Oh, my bad. I thought this was LDK?

**Reply by okjodom** [reply]
We have a [Fedimint gateway  lightning service spec](https://github.com/fedimint/fedimint/blob/master/gateway/ln-gateway/proto/gatewaylnrpc.proto) that we'll be looking to implement for LDK nodes especially in server scenarios.

At the moment, this spec is only implemented [as an extension for CLN nodes](https://github.com/fedimint/fedimint/blob/master/gateway/ln-gateway/src/bin/cln_extension.rs), and  [for LND nodes](https://github.com/fedimint/fedimint/blob/master/gateway/ln-gateway/src/lnd.rs).

**Reply by moneyball** [reply]
Sure but there is no active development by the Fedimint project to use LDK.

The bar for being on this list is at least 1 developer actively building with LDK.

**Reply by benthecarman** [reply]
I think the library only used for creating invoices

**Reply by okjodom** [reply]
@EthnTuttle , your references are the dependencies we have on lightning libraries from LDK, but it's different from what the integration would be when we start building with LDK

**Reply by EthnTuttle** [reply]
ack! Thnx for the clarification.

**Reply by moneyball** [reply]
So Fedimint uses LDK's code to create invoices? If so then that qualifies for "building with LDK."

**Reply by okjodom** [reply]
Ah yes, then we've passed the bar! For some reason, I've been thinking we'll qualify when we are able to have LDK as the node implementation sitting between a Fedimint gateway and the Lightning network. These dependencies alone didn't seem enough because they should be natural first choice dependencies in any rust project doing lightning today.

**Reply by moneyball** [reply]
> they should be natural first choice dependencies in any rust project doing lightning today.

I'm not clear on why this would disqualify this as "building with LDK" then? VLS, TEOS, and LNDK projects only use portions of LDK for their needs, but they're still getting value out of the LDK project, which to me is why it should count if Fedimint does use the LDK code/library for invoices. Maybe because this repo was "imported" into LDK from another non/predated-LDK developer it shouldn't count?

**Comment by ConorOkus** [comment]
https://github.com/aniketambore/bijli_ln_wallet

**Comment by ahrefs13** [comment]
https://workit.fitness
